rosbag2_cpp/src/rosbag2_cpp/typesupport_helpers.cpp

```diff
@@ -105,33 +105,51 @@ get_typesupport(const std::string & type, const std::string & typesupport_identi
   std::string type_name;
   std::tie(package_name, middle_module, type_name) = extract_type_identifier(type);
 
-  std::string poco_dynamic_loading_error = "Something went wrong loading the typesupport library "
+  std::string rcutils_dynamic_loading_error =
+    "Something went wrong loading the typesupport library "
     "for message type " + package_name + "/" + type_name + ".";
 
   auto library_path = get_typesupport_library_path(package_name, typesupport_identifier);
 
-  try {
-    auto typesupport_library = std::make_shared<Poco::SharedLibrary>(library_path);
-
-    auto symbol_name = typesupport_identifier + "__get_message_type_support_handle__" +
-      package_name + "__" + (middle_module.empty() ? "msg" : middle_module) + "__" + type_name;
-
-    if (!typesupport_library->hasSymbol(symbol_name)) {
-      throw std::runtime_error(poco_dynamic_loading_error + " Symbol not found.");
+  rcutils_shared_library_t * typesupport_library = nullptr;
+
+  typesupport_library = new rcutils_shared_library_t;
```

Comment:
We shouldn't be using `new` and `delete` in modern C++. I think you can use a `unique_ptr` just fine here, and then don't have to add the `delete` calls in the error cases.